File: lab8_2.js
function countPrimes(n) {
    if (n < 2) return 0;
    const isPrime = Array(n).fill(true);
    isPrime[0] = isPrime[1] = false;
    for (let i = 2; i * i < n; i++) {
        if (isPrime[i]) {
            for (let j = i * i; j < n; j += i) {
                isPrime[j] = false;
            }
        }
    }
    return isPrime.filter(Boolean).length;
}

// Unit test
const assert = require('assert');
assert.strictEqual(countPrimes(18), 7);
assert.strictEqual(countPrimes(10), 4);
assert.strictEqual(countPrimes(2), 0);
assert.strictEqual(countPrimes(1), 0);

console.log("All tests passed!");

File: lab8_2.py
def countPrimes(n):
    if n < 2:
        return 0
    is_prime = [True] * n
    is_prime[0] = is_prime[1] = False
    for i in range(2, int(n**0.5) + 1):
        if is_prime[i]:
            for j in range(i * i, n, i):
                is_prime[j] = False
    return sum(is_prime)

import unittest
# Unit test
class TestCountPrimes(unittest.TestCase):
    def test_count_primes(self):
        self.assertEqual(countPrimes(18), 7)
        self.assertEqual(countPrimes(10), 4)
        self.assertEqual(countPrimes(2), 0)
        self.assertEqual(countPrimes(1), 0)

if __name__ == "__main__":
    unittest.main()
